### ISSUE TITLE
Device Form Validation with Device Type & Platform

### ISSUE THREAD

**Issue by itdependsnetworks** [nautobot/nautobot#4931]
### Environment
* Nautobot version (Docker tag too if applicable): 2.0.5
* Python version: 3.11
* Database platform, version: N/A
* Middleware(s): N/A

### Steps to Reproduce
1. Create manufacturer "Cisco" 
2. Create Platform "Cisco IOS", assign to manufacturer "Cisco"
3. Create Device Type "CSR1000V", assign to manufacturer "Cisco"
4. Create Device "NYC-RT01", assing to CSR1000V and Cisco IOS
5. Attempt to change to Arista

### Expected Behavior

Form validates and allows

### Observed Behavior

Either fails validation or have to remove platform and edit it at second time.

**Comment by itdependsnetworks** [comment]
I think this logic needs to be updated https://github.com/nautobot/nautobot/blob/8629dfbcb2478a3db3162c7c3c600efba6f5b42b/nautobot/dcim/forms.py#L1604

**Comment by gsnider2195** [comment]
You're not able to change the manufacturer, device type and platform in a single edit form because the `self.fields["platform"].queryset` is populated from the existing instance's `device_type`, not the current `device_type` selected in the form.

**Comment by itdependsnetworks** [comment]
> is populated from the existing instance's device_type, not the current device_type selected in the form.

Yes, that is what my suggested code in the comment before yours attempts to address. I am not so sure that `self.fields["device_type"].manufacturer` will actually work, but that is the intention of it.
